### PR TITLE
Update conflict constraint for "nelmio/api-doc-bundle" in order to allow version 3

### PR DESCRIPTION
**Pull request by wbloszyk**
## Subject
Update conflict constraint for "nelmio/api-doc-bundle" in order to allow version 3.

I am targeting this branch, because these changes almost respect BC (with one, very little BC-break potential).

## Changelog

```markdown
### Added
- support for `nelmio/api-doc-bundle` >= 3.6
### Changed
- Updated docs in order to expose how to configure custom serialization paths under jms_serializer configuration node (jms_serializer.metadata.directories).
```

### CODE REVIEW

File: src/Resources/config/command.xml [hunk 1]
```diff
@@ -1,6 +1,9 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <container xmlns="http://symfony.com/schema/dic/services" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
     <services>
+        <service id="Sonata\NewsBundle\Command\SwaggerDocblockConvertCommand" class="Sonata\NewsBundle\Command\SwaggerDocblockConvertCommand" public="true">
```
Comment (phansys) [inline src/Resources/config/command.xml:4]: IMO, this command doesn't belong to this package.

Comment (wbloszyk) [reply src/Resources/config/command.xml:4]: I use it to convert API. My bad.

File: src/Controller/Api/Legacy/PostController.php [hunk 2]
```diff
@@ -0,0 +1,460 @@
...
+    {
+        $criteria = $paramFetcher->all();
+
+        unset($criteria['page'], $criteria['count']);
+
+        foreach ($criteria as $key => $value) {
+            if (null === $value) {
+                unset($criteria[$key]);
+            }
+        }
```
Comment (phansys) [inline src/Controller/Api/Legacy/PostController.php:383]:
```suggestion
        $criteria = array_filter($criteria, static function ($value): bool {
            return null !== $value;
        });
```

**Review by phansys — CHANGES_REQUESTED** [review]

File: src/Controller/Api/PostController.php [hunk 3]
```diff
@@ -243,27 +340,33 @@ public function getPostCommentsAction($id, ParamFetcherInterface $paramFetcher)
         $criteria = $this->filterCriteria($paramFetcher);
         $criteria['postId'] = $post->getId();
 
-        /** @var PagerInterface $pager */
-        $pager = $this->commentManager->getPager($criteria, $page, $count);
 
-        return $pager;
+        return  $this->commentManager->getPager($criteria, $page, $count);
```
Comment (phansys) [inline src/Controller/Api/PostController.php:344]:
```suggestion
        return $this->commentManager->getPager($criteria, $page, $count);
```

File: src/Controller/Api/Legacy/PostController.php [hunk 4]
```diff
@@ -0,0 +1,453 @@
...
+        $criteria = $paramFetcher->all();
+
+        unset($criteria['page'], $criteria['count']);
+
+        $criteria = array_filter($criteria, static function ($value): bool {
+            return null !== $value;
+        });
+
+        if (\array_key_exists('dateValue', $criteria)) {
+            $date = new \DateTime($criteria['dateValue']);
```
Comment (phansys) [inline src/Controller/Api/Legacy/PostController.php:379]: Maybe we could use `\DateTimeImmutable` here.

**Review by phansys — CHANGES_REQUESTED** [review]

File: src/Controller/Api/Legacy/PostController.php (hunk 4 above)
Comment (wbloszyk) [reply src/Controller/Api/Legacy/PostController.php:379]: I'm not sure. IMO we should not do deep refactoring.

**Review by phansys — DISMISSED** [review]

**Review by phansys — DISMISSED** [review]

File: tests/App/Resources/config/sonata.yaml [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    title: Sonata Project
+    link: https://sonata-project.org
+    description: Cool bundles on top of Symfony
+    salt: 'secureToken'
+    permalink_generator: sonata.news.permalink.date
+    db_driver: 'doctrine_orm'
+    comment:
+        notification:
+            emails: [email@example.org, email2@example.org]
+            from: no-reply@sonata-project.org
```
Comment (phansys) [inline tests/App/Resources/config/sonata.yaml:11]:
```suggestion
            emails: ['email@example.org', 'email2@example.org']
            from: 'no-reply@sonata-project.org'
```

**Review by phansys — CHANGES_REQUESTED** [review]

File: tests/App/Resources/config/routing/routes.yaml [hunk 6]
```diff
@@ -0,0 +1,3 @@
+test_page:
+    path: /u_truncate_test
+    controller: Sonata\NewsBundle\Tests\App\Controller\TruncateController:test
```
Comment (phansys) [inline tests/App/Resources/config/routing/routes.yaml:3]:
```suggestion
    controller: Sonata\NewsBundle\Tests\App\Controller\TruncateController::test
```

File: tests/App/AppKernel.php [hunk 7]
```diff
@@ -87,17 +94,23 @@ public function getProjectDir()
...
     {
-        $loader->load($this->getProjectDir().'/config/config.yaml');
+        $loader->load($this->getProjectDir().'/Resources/config/config.yaml');
         $containerBuilder->setParameter('app.base_dir', $this->getBaseDir());
     }
 
     private function getBaseDir(): string
     {
-        return sys_get_temp_dir().'/sonata-news-bundle/var/';
+        return sys_get_temp_dir().'/sonata-news-bundle/var/dase12r1aa23r213';
```
Comment (phansys) [inline tests/App/AppKernel.php:114]: :eyes:

**Review by phansys — CHANGES_REQUESTED** [review]

**Review by phansys — DISMISSED** [review]

**Comment by OskarStark** [comment]
@wbloszyk can we close #604 ?

**Comment by wbloszyk** [comment]
> @wbloszyk can we close #604 ?

Closed. I should find time to end it in this week.

**Comment by VincentLanglet** [comment]
What's the status of this PR ? Btw, tests are failing

**Comment by jordisala1991** [comment]
Abandoned package.

**Pull request closed without merging**